mne_bids/inspect.py

```diff
@@ -0,0 +1,128 @@
+             show_options=True, block=True)
+    updated_bads = raw.info['bads']
+
+    if set(orig_bads) == set(updated_bads):
+        # Nothing has changed, so we can just exit.
+        return
+
+    _save_bads_dialog_box(bads=updated_bads, bids_path=bids_path,
+                          verbose=verbose)
+
+
+def _save_bads_dialog_box(*, bads, bids_path, verbose):
+    """Display a dialog box asking whether to save the changes."""
+    title = 'Save changes?'
+    message = (f'You have modified the bad channel selection of\n'
+               f'{bids_path.basename}.\n\nWould you like to save these '
+               f'changes to the\nBIDS dataset?')
+    icon_fname = Path(__file__).parent / 'assets' / 'help-128px.png'
+    icon = plt.imread(icon_fname)
+
+    fig = plt.figure(figsize=(7.5, 2))
+    fig.canvas.toolbar.setVisible(False)  # This only works with QtAgg.
+    fig.canvas.set_window_title('MNE-BIDS Inspector')
+    fig.suptitle(title, y=0.95, fontsize='xx-large', fontweight='bold')
+
+    gs = fig.add_gridspec(1, 2, width_ratios=(1.5, 5))
+    ax_icon = fig.add_subplot(gs[0, 0])
+    ax_msg = fig.add_subplot(gs[0, 1])
+
+    ax_icon.imshow(icon)
+    ax_icon.axis('off')
+
+    ax_msg.text(x=0, y=0.8, s=message, fontsize='large',
+                verticalalignment='top', horizontalalignment='left',
+                multialignment='left')
+    ax_msg.axis('off')
+
+    ax_save = plt.axes([0.6, 0.05, 0.3, 0.1])
+    ax_dont_save = plt.axes([0.1, 0.05, 0.3, 0.1])
+
+    save_button = Button(ax=ax_save, label='Save')
+    save_button.label.set_fontsize('medium')
+    save_button.label.set_fontweight('bold')
+
+    dont_save_button = Button(ax=ax_dont_save, label="Don't save")
+    dont_save_button.label.set_fontsize('medium')
+    dont_save_button.label.set_fontweight('bold')
+
+    def _save_callback(event):
+        plt.close(fig)
+        _save_bads(bads=bads, bids_path=bids_path, verbose=verbose)
+
+    def _dont_save_callback(event):
+        plt.close(fig)
+
+    save_button.on_clicked(_save_callback)
+    dont_save_button.on_clicked(_dont_save_callback)
+    plt.show(block=True)
+
+
+def _save_bads(*, bads, bids_path, verbose):
+    """Update the set of channels marked as bad in the sidecar."""
+    channels_tsv_fname = bids_path.copy().update(suffix='channels',
+                                                 extension='.tsv')
+    channels_tsv_data = _from_tsv(channels_tsv_fname)
+
+    descriptions = []
+    for ch_name in bads:
+        idx = channels_tsv_data['name'].index(ch_name)
+        if channels_tsv_data['status'][idx] == 'bad':
+            # Channel was already marked as bad, retain existing description.
+            description = channels_tsv_data['status_description'][idx]
+        else:
+            # Channel has been manually marked as bad during inspection, assign
+            # default description.
+            description = 'Manual inspection via MNE-BIDS'
```

Comment:
I wonder if we can go a step further and say by who using

`os.getenv('USER')`, which would work for everyone except Windows users(?).

Reply:
How about
```python
In [5]: Path(os.path.expanduser('~')).name
Out[5]: 'hoechenberger'
```
And we could even add a timestamp.

Reply:
… but maybe that's difficult in terms of users' privacy. We should at least not do that without very explicitly letting them know about it.

Reply:
I like the idea of a timestamp! Hmm I suppose, but perhaps `user` can be a kwarg of the function then, so that way it is explicit, else it defaults to the behavior you have here?